src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,277 @@
+        Scanner fileScanner = new Scanner(file);
+        while (fileScanner.hasNext()) {
+            String line = fileScanner.nextLine();
+            String otherThanQuote = " [^\"] ";
+            String quotedString = String.format(" \" %s* \" ", otherThanQuote);
+            String regex = String.format("(?x) " + // enable comments, ignore white spaces
+                            ",                         " + // match a comma
+                            "(?=                       " + // start positive look ahead
+                            "  (?:                     " + //   start non-capturing group 1
+                            "    %s*                   " + //     match 'otherThanQuote' zero or more times
+                            "    %s                    " + //     match 'quotedString'
+                            "  )*                      " + //   end group 1 and repeat it zero or more times
+                            "  %s*                     " + //   match 'otherThanQuote'
+                            "  $                       " + // match the end of the string
+                            ")                         ", // stop positive look ahead
+                    otherThanQuote, quotedString, otherThanQuote);
+            String[] args = line.split(regex, -1);
+            try {
+                boolean isMarked = Integer.parseInt(args[1]) == 1;
+                for (int i = 2; i < args.length; ++i) {
+                    args[i] = args[i].substring(1, args[i].length()-1);
+                }
+                String content = args[2];
+                switch (args[0]) {
+                case "todo":
+                    taskList.add(new Todo(content, isMarked));
+                    break;
+                case "deadline":
+                    taskList.add(new Deadline(content, args[3], isMarked));
+                    break;
+                case "event":
+                    taskList.add(new Event(content, args[3], args[4], isMarked));
+                    break;
+                default:
+                    throw new IllegalArgumentException();
+                }
+            } catch (IllegalArgumentException e) {
+                System.err.println("Cannot add task: " + line);
+            }
+        }
+    }
+
+    private static void writeData(String fileName) {
+        try {
+            FileWriter fw = new FileWriter(fileName);
+            for (Task task : taskList) {
+                fw.write(task.toCSV() + System.lineSeparator());
+            }
+            fw.close();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void main(String[] args) {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
```

Comment:
Code chunks such as this can be abstracted into a single method call to maintain a similar level of abstraction across the method (in this case, your switch statements). This will help enhance readability of your code.